pkg/aws/aws.go

```diff
@@ -115,6 +119,24 @@ func New(ctx context.Context, config *Config) (*AWS, error) {
 			}, awsClient)
 			// TODO: append new aws rds collectors next
 			// collectors = append(collectors, collector)
+		case "NATGATEWAY":
+			// Build a Cost Explorer client using the same region/profile settings
+			loadOpts := []func(*awsconfig.LoadOptions) error{
+				awsconfig.WithEC2IMDSRegion(),
+			}
+			if config.Profile != "" {
+				loadOpts = append(loadOpts, awsconfig.WithSharedConfigProfile(config.Profile))
+			}
+			if config.Region != "" {
+				loadOpts = append(loadOpts, awsconfig.WithRegion(config.Region))
+			}
+			acfg, err := awsconfig.LoadDefaultConfig(ctx, loadOpts...)
+			if err != nil {
+				return nil, fmt.Errorf("failed to load AWS config for NAT Gateway: %w", err)
+			}
+			ceClient := awscostexplorer.NewFromConfig(acfg)
+			gwCollector := awsgwnat.New(config.ScrapeInterval, ceClient)
+			collectors = append(collectors, gwCollector)
```

Comment:
Please update to the new client initialization method - it should abstract a lot of this away 😄  for an example, [see `case serviceEC2` on `main`](https://github.com/grafana/cloudcost-exporter/blob/main/pkg/aws/aws.go#L120-L127)

Reply:
for now I've just rebase-d it at 4261dc4  (essentially to make it work again and void losing functionality), will see how to use new multi-cli stuff

Reply:
hopef00ly addressed at c7bc4ce